Common/src/main/java/com/blamejared/crafttweaker/api/util/sequence/task/ISequenceTask.java

```diff
@@ -0,0 +1,44 @@
+package com.blamejared.crafttweaker.api.util.sequence.task;
+
+import com.blamejared.crafttweaker.api.annotation.ZenRegister;
+import com.blamejared.crafttweaker.api.util.sequence.SequenceContext;
+import com.blamejared.crafttweaker_annotations.annotations.Document;
+import org.openzen.zencode.java.ZenCodeType;
+
+/**
+ * A task in a {@link com.blamejared.crafttweaker.api.util.sequence.Sequence}.
+ *
+ * @docParam this new SleepTask(20)
+ */
+@ZenRegister
+@Document("vanilla/api/sequence/task/ISequenceTask")
+@ZenCodeType.Name("crafttweaker.api.sequence.task.ISequenceTask")
+public interface ISequenceTask<T, U> {
+    
+    /**
+     * Ticks this task.
+     *
+     * <p>A check to {@code isComplete} should be done before ticking!.</p>
```

Comment:
Would it be worth to wrap tasks into one that checks if it's complete and if so does nothing at registration time? Something similar to

```java
class CheckingTask implements Task {
  private Task wrapped;
  public void tick() {
    if (wrapped.complete) return;
    wrapped.tick();
  } 
}
```